File: react-app/src/store/price.js
const ALL_PRICES = "price/ALL_PRICES"
const ADD_PRICE = "price/ADD_PRICE"
const ONE_PRICE = "price/ONE_PRICE"
const DELETE_PRICE = "post/DELETE_PRICE";

const allPrices = (payload) => ({
    type: ALL_PRICES,
    payload,
})

const addPrice = (payload) => ({
    type: ADD_PRICE,
    payload
})

const singlePrice = (payload) => ({
  type: ONE_PRICE,
  payload,
});

const deletePrice = (payload) => ({
  type: DELETE_PRICE,
  payload,
});



//GET ALL PRICES
export const thunk_getAllPrices = () => async (dispatch) => {
  const res = await fetch(`/api/prices/`);

  if (res.ok) {
    const prices = await res.json();
    dispatch(allPrices(prices));
    return prices;
  }
};

//GET ALL PRICES
export const thunk_getAPI = () => async (dispatch) => {
  const res = await fetch(`/api/api/`);

  if (res.ok) {
    const prices = await res.json();
    // dispatch(allSpots(spots));
    return prices;
  }
};


//GET ONE PRICE
export const thunk_getOnePrice = (id) => async (dispatch) => {
  const res = await fetch(`/api/prices/${id}`);

  if (res.ok) {
    const price = await res.json();
    dispatch(singlePrice(price));
    return price;
  }
};

//ADD A SPOT
export const thunk_addPrice =
  ({ userId, varOne, varTwo, varThree, varFour, varFive, varSix, varSeven, varEight, varNine, varTen, varEleven, varTwelve, varThirteen, varFourteen, varFifteen, varSixteen, varSeventeen, varEighteen, varNinteen, varTwenty, varTwentyOne, varTwentyTwo, varTwentyThree, varTwentyFour, varTwentyFive, varTwentySix, varTwentySeven, varTwentyEight, varTwentyNine, varThirty, varThirtyOne, varThirtyTwo, varThirtyThree, varThirtyFour, varThirtyFive, varThirtySix, varThirtySeven, varThirtyEigh, varThirtyNine, varFourty, varFourtyOne, varFourtyTwo, varFourtyThree, varFourtyFour, varFourtyFive, varFourtySix, varFourtySeven, varFourtyEight, varFourtyNine, varFifty, varFiftyOne, varFiftyTwo, varFiftyThree, varFiftyFour, varFiftyFive, varFiftySix, varFiftySeven, varFiftyEight,}) =>
  async (dispatch) => {
    const res = await fetch("/api/prices/", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        userId,
        varOne,
        varTwo,

        varThree,
        varFour,

        varFive,
        varSix,
        varSeven, varEight, varNine, varTen, varEleven, varTwelve, varThirteen, varFourteen, varFifteen, varSixteen, varSeventeen, varEighteen, varNinteen, varTwenty, varTwentyOne, varTwentyTwo, varTwentyThree, varTwentyFour, varTwentyFive, varTwentySix, varTwentySeven, varTwentyEight, varTwentyNine, varThirty, varThirtyOne, varThirtyTwo, varThirtyThree, varThirtyFour, varThirtyFive, varThirtySix, varThirtySeven, varThirtyEigh, varThirtyNine, varFourty, varFourtyOne, varFourtyTwo, varFourtyThree, varFourtyFour, varFourtyFive, varFourtySix, varFourtySeven, varFourtyEight, varFourtyNine, varFifty, varFiftyOne, varFiftyTwo, varFiftyThree, varFiftyFour, varFiftyFive, varFiftySix, varFiftySeven, varFiftyEight,
      }),
    });

    if (res.ok) {
      const price = await res.json();
      dispatch(addPrice(price));
      return price;
    }
  };

  //UPDATE PRICE
  export const thunk_updatePrice =
    ({ id, userId,  varOne, varTwo, varThree, varFour, varFive, varSix, varSeven, varEight, varNine, varTen, varEleven, varTweleve,  varThirteen, varFourteen, varFifteen, varSixteen, varSeventeen, varEighteen, varNinteen, varTwenty, varTwentyOne, varTwentyTwo, varTwentyThree, varTwentyFour, varTwentyFive, varTwentySix, varTwentySeven, varTwentyEight, varTwentyNine, varThirty, varThirtyOne, varThirtyTwo, varThirtyThree, varThirtyFour, varThirtyFive, varThirtySix, varThirtySeven, varThirtyEigh, varThirtyNine, varFourty, varFourtyOne, varFourtyTwo, varFourtyThree, varFourtyFour, varFourtyFive, varFourtySix, varFourtySeven, varFourtyEight, varFourtyNine, varFifty, varFiftyOne, varFiftyTwo, varFiftyThree, varFiftyFour, }) =>
    async (dispatch) => {

      const res = await fetch(`/api/prices/${id}/edit`, {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          userId,
          varOne, varTwo, varThree, varFour, varFive, varSix, varSeven, varEight, varNine, varTen, varEleven, varTweleve,  varThirteen, varFourteen, varFifteen, varSixteen, varSeventeen, varEighteen, varNinteen, varTwenty, varTwentyOne, varTwentyTwo, varTwentyThree, varTwentyFour, varTwentyFive, varTwentySix, varTwentySeven, varTwentyEight, varTwentyNine, varThirty, varThirtyOne, varThirtyTwo, varThirtyThree, varThirtyFour, varThirtyFive, varThirtySix, varThirtySeven, varThirtyEigh, varThirtyNine, varFourty, varFourtyOne, varFourtyTwo, varFourtyThree, varFourtyFour, varFourtyFive, varFourtySix, varFourtySeven, varFourtyEight, varFourtyNine, varFifty, varFiftyOne, varFiftyTwo, varFiftyThree, varFiftyFour,

        }),
      });

      if (res.ok) {
        const price = await res.json();
        dispatch(addPrice(price));
        return price;
      }
    };

  // DELETE PRICE
  export const thunk_deletePrice = ({ id }) => async (dispatch) => {
    const res = await fetch(`/api/prices/${id}`, {
      method: 'DELETE',
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify({
        id
      })
    });

    if (res.ok) {
      const deletedPrice = await res.json();
      dispatch(deletePrice(deletedPrice));
      return "Deletion successful";
    }
  };



//SPOT REDUCER
const priceReducer = (state = {}, action) => {
  switch (action.type) {
    case ALL_PRICES: {
      const newState = { ...state };
      newState["allPrices"] = action.payload;
      return newState;
    }
    case ONE_PRICE: {
      const newState = { ...state };
      newState["onePrice"] = action.payload;
      return newState;
    }
    case DELETE_PRICE: {
      const newState = { ...state };
      delete newState[action.payload];
      return newState;
    }
    default:
      return state;
  }
};

export default priceReducer;

File: app/models/price.py
from .db import db
from sqlalchemy import DateTime
import datetime

class Price(db.Model):
    __tablename__ = 'prices'

    id = db.Column(db.Integer, primary_key=True)
    varOne = db.Column(db.Integer, nullable=True)
    varTwo = db.Column(db.Integer, nullable=True)
    varThree = db.Column(db.Integer, nullable=True)
    varFour = db.Column(db.Integer, nullable=True)
    varFive = db.Column(db.Integer, nullable=True)
    varSix = db.Column(db.Integer, nullable=True)
    varSeven = db.Column(db.Integer, nullable=True)
    varEight = db.Column(db.Integer, nullable=True)
    varNine = db.Column(db.Integer, nullable=True)
    varTen = db.Column(db.Integer, nullable=True)
    varEleven = db.Column(db.Integer, nullable=True)
    varTwelve = db.Column(db.Integer, nullable=True)
    varThirteen = db.Column(db.Integer, nullable=True)
    varFourteen = db.Column(db.Integer, nullable=True)
    varFifteen = db.Column(db.Integer, nullable=True)
    varSixteen = db.Column(db.Integer, nullable=True)
    varSeventeen = db.Column(db.Integer, nullable=True)
    varEighteen = db.Column(db.Integer, nullable=True)
    varNinteen = db.Column(db.Integer, nullable=True)
    varTwenty = db.Column(db.Integer, nullable=True)
    varTwentyOne = db.Column(db.Integer, nullable=True)
    varTwentyTwo = db.Column(db.Integer, nullable=True)
    varTwentyThree = db.Column(db.Integer, nullable=True)
    varTwentyFour = db.Column(db.Integer, nullable=True)
    varTwentyFive = db.Column(db.Integer, nullable=True)
    varTwentySix = db.Column(db.Integer, nullable=True)
    varTwentySeven = db.Column(db.Integer, nullable=True)
    varTwentyEight = db.Column(db.Integer, nullable=True)
    varTwentyNine = db.Column(db.Integer, nullable=True)
    varThirty = db.Column(db.Integer, nullable=True)
    varThirtyOne = db.Column(db.Integer, nullable=True)
    varThirtyTwo = db.Column(db.Integer, nullable=True)
    varThirtyThree = db.Column(db.Integer, nullable=True)
    varThirtyFour = db.Column(db.Integer, nullable=True)
    varThirtyFive = db.Column(db.Integer, nullable=True)
    varThirtySix = db.Column(db.Integer, nullable=True)
    varThirtySeven = db.Column(db.Integer, nullable=True)
    varThirtyEigh = db.Column(db.Integer, nullable=True)
    varThirtyNine = db.Column(db.Integer, nullable=True)
    varFourty = db.Column(db.Integer, nullable=True)
    varFourtyOne = db.Column(db.Integer, nullable=True)
    varFourtyTwo = db.Column(db.Integer, nullable=True)
    varFourtyThree = db.Column(db.Integer, nullable=True)
    varFourtyFour = db.Column(db.Integer, nullable=True)
    varFourtyFive = db.Column(db.Integer, nullable=True)
    varFourtySix = db.Column(db.Integer, nullable=True)
    varFourtySeven = db.Column(db.Integer, nullable=True)
    varFourtyEight = db.Column(db.Integer, nullable=True)
    varFourtyNine = db.Column(db.Integer, nullable=True)
    varFifty = db.Column(db.Integer, nullable=True)
    varFiftyOne = db.Column(db.Integer, nullable=True)
    varFiftyTwo = db.Column(db.Integer, nullable=True)
    varFiftyThree = db.Column(db.Integer, nullable=True)
    varFiftyFour = db.Column(db.Integer, nullable=True)
    varFiftyFive = db.Column(db.Integer, nullable=True)
    varFiftySix = db.Column(db.Integer, nullable=True)
    varFiftySeven = db.Column(db.Integer, nullable=True)
    varFiftyEight = db.Column(db.Integer, nullable=True)
    created_at = db.Column(DateTime, default=datetime.datetime.utcnow)
    updated_at = db.Column(DateTime, default=datetime.datetime.utcnow)




    userId = db.Column(db.Integer, db.ForeignKey('users.id'))
    user = db.relationship("User", back_populates="prices")



    def to_dict(self):
        return {
            "id": self.id,
            "varOne" : self.varOne,
            "varTwo" : self.varTwo,
            "varThree" : self.varThree,
            "varFour" : self.varFour,
            "varFive" : self.varFive,
            "varSix" : self.varSix,
            "varSeven" : self.varSeven,
            "varEight" : self.varEight,
            "varNine" : self.varNine,
            "varTen" : self.varTen,
            "varEleven" : self.varEleven,
            "varTwelve" : self.varTwelve,
            "varThirteen" : self.varThirteen,
            "varFourteen" : self.varFourteen,
            "varFifteen" : self.varFifteen,
            "varSixteen" : self.varSixteen,
            "varSeventeen" : self.varSeventeen,
            "varEighteen" : self.varEighteen,
            "varNinteen" : self.varNinteen,
            "varTwenty" : self.varTwenty,
            "varTwentyOne" : self.varTwentyOne,
            "varTwentyTwo" : self.varTwentyTwo,
            "varTwentyThree" : self.varTwentyThree,
            "varTwentyFour" : self.varTwentyFour,
            "varTwentyFive" : self.varTwentyFive,
            "varTwentySix" : self.varTwentySix,
            "varTwentySeven" : self.varTwentySeven,
            "varTwentyEight" : self.varTwentyEight,
            "varTwentyNine" : self.varTwentyNine,
            "varThirty" : self.varThirty,
            "varThirtyOne" : self.varThirtyOne,
            "varThirtyTwo" : self.varThirtyTwo,
            "varThirtyThree" : self.varThirtyThree,
            "varThirtyFour" : self.varThirtyFour,
            "varThirtyFive" : self.varThirtyFive,
            "varThirtySix" : self.varThirtySix,
            "varThirtySeven" : self.varThirtySeven,
            "varThirtyEight" : self.varThirtyEigh,
            "varThirtyNine" : self.varThirtyNine,
            "varFourty" : self.varFourty,
            "varFourtyOne" : self.varFourtyOne,
            "varFourtyTwo" : self.varFourtyTwo,
            "varFourtyThree" : self.varFourtyThree,
            "varFourtyFour" : self.varFourtyFour,
            "varFourtyFive" : self.varFourtyFive,
            "varFourtySix" : self.varOne,
            "varFourtySeven" : self.varFourtySeven,
            "varFourtyEight" : self.varFourtyEight,
            "varFourtyNine" : self.varFourtyNine,
            "varFifty" : self.varFifty,
            "varFiftyOne" : self.varFiftyOne,
            "varFiftyTwo" : self.varFiftyTwo,
            "varFiftyThree" : self.varFiftyThree,
            "varFiftyFour" : self.varFiftyFour,
            "varFiftyFive" : self.varFiftyFive,
            "varFiftySix" : self.varFiftySix,
            "varFiftySeven" : self.varFiftySeven,
            "varFiftyEight" : self.varFiftyEight

        }
